invenio_logging/sentry.py

```diff
@@ -55,28 +56,61 @@ def init_config(self, app):
 
     def install_handler(self, app):
         """Install log handler."""
-        from raven.conf import EXCLUDE_LOGGER_DEFAULTS
-        from raven.contrib.celery import register_logger_signal, \
-            register_signal
-        from raven.contrib.flask import Sentry, make_client
-        from raven.handlers.logging import SentryHandler
-
-        # Installs sentry in app.extensions['sentry']
         level = getattr(logging, app.config['LOGGING_SENTRY_LEVEL'])
-
         logging_exclusions = None
         if not app.config['LOGGING_SENTRY_PYWARNINGS']:
-            logging_exclusions = list(EXCLUDE_LOGGER_DEFAULTS)
-            logging_exclusions.append('py.warnings')
+            logging_exclusions = (
+                'raven',
+                'gunicorn',
+                'south',
+                'sentry.errors',
+                'django.request',
+                'dill',
+                'py.warnings')
+        if app.config['SENTRY_SDK']:
+            self.install_sentry_sdk_handler(app, logging_exclusions, level)
+        else:
+            self.install_raven_handler(app, logging_exclusions, level)
+
+        # Werkzeug only adds a stream handler if there's no other handlers
+        # defined, so when Sentry adds a log handler no output is
+        # received from Werkzeug unless we install a console handler
+        # here on the werkzeug logger.
+        if app.debug:
+            logger = logging.getLogger('werkzeug')
+            logger.setLevel(logging.INFO)
+            logger.addHandler(logging.StreamHandler())
+
+    def install_sentry_sdk_handler(self, app, logging_exclusions, level):
+        """Install sentry-python sdk log handler."""
+        import sentry_sdk
+        from sentry_sdk.integrations.flask import FlaskIntegration
+        from sentry_sdk import configure_scope
+
+        sentry_sdk.init(
+            dsn=app.config['SENTRY_DSN'],
+            in_app_exclude=logging_exclusions,
+            integrations=[FlaskIntegration()],
```

Comment:
How is the Celery integration handled? https://docs.sentry.io/platforms/python/celery/

Reply:
I can't make it work with celery.

I am keep getting the following error.

```
The full contents of the message body was:
b'\x93\x92\x02\x02\x80\x84\xa9callbacks\xc0\xa8errbacks\xc0\xa5chain\xc0\xa5chord\xc0' (41b)
Traceback (most recent call last):
  File "/opt/rh/rh-python36/root/usr/lib/python3.6/site-packages/celery/worker/consumer/consumer.py", line 558, in on_task_received
    strategy = strategies[type_]
KeyError: 'my_site.tasks.sum'
```

Reply:
<img width="683" alt="Screenshot 2019-08-02 at 09 00 29" src="https://user-images.githubusercontent.com/5921114/62350954-907b3100-b504-11e9-9988-7e42b6810dfc.png">


Works with celery as well.